executor/cook/server.py

```diff
@@ -0,0 +1,109 @@
+"""
+An HTTP server exposing a simple API for updating task data. It is meant to be consumed by
+the processes of a task's commands. It allows setting custom failure messages and updating
+task progress.
+"""
+
+import re
+import json
+import socket
+import logging
+
+from threading import Thread
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+class CookExecutorHTTPRequestHandler(BaseHTTPRequestHandler):
+    """
+    A BaseHTTPRequestHandler subclass implementing a very simple HTTP API
+    """
+
+    def do_GET(self):
+        """
+        GET /health
+          => returns 200
+        GET /task/:task_id
+          => returns 200 and task data if task_id is valid
+          => returns 404 if task_id is unknown or invalid
+        """
+        if self.path is '/health':
+            self.send_json_response(200, {'response': 'okay'})
+        elif re.match('\/task\/[^/]+', self.path):
+            task_id = self.path.split('/')[2]
+            task_data = self.server.store.get('task', task_id)
+
+            if task_data:
+                self.send_json_response(200, task_data)
+            else:
+                self.send_json_response(404, {'response': 'not found'})
+
+    def do_PATCH(self):
```

Comment:
The documentation specifies that PATCH can update certain keys:

> Since it is a `PATCH` request, a valid request may contain only some of the following keys: ... progress ... message ... env

Where is it enforced that only those keys may be updated?